### PR TITLE
Follow the pointer during decomposition

### PR DESCRIPTION
**Pull request by mikes-blend**
Memory addresses aren't useful Strings.

### CODE REVIEW

**Comment by mikes-blend** [comment]
Need to handle nil better

**Comment by mikes-blend** [comment]
Tested that nil behavior is now consistent with go-commons for better or worse.

File: util/reflection.go [hunk 1]
```diff
@@ -393,13 +393,26 @@ func (ru reflectionUtil) DecomposeStrings(obj interface{}, tagName ...string) ma
...
-			output[dataField] = fmt.Sprintf("%v", fieldValue.Interface())
+			output[dataField] = fmt.Sprintf("%v", ru.followValuePointer(fieldValue))
 		}
 	}
 
 	return output
 }
 
+// followValuePointer derefs a reflectValue until it isn't a pointer, but will preseve it's nilness.
+func (ru reflectionUtil) followValuePointer(v reflect.Value) interface{} {
```
Comment (wcharczuk) [inline util/reflection.go]: did this function really not already exist?

Comment (amccarthy1) [reply util/reflection.go]: https://github.com/blend/go-sdk/blob/cf0e4b34712516647c7be247ee1c4a8725441ab6/util/reflection.go#L43-L54

I think it does...

**Pull request merged**